projects/scannet_offline_eval/eval/refer_exp.py

```diff
@@ -0,0 +1,264 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+"""
+    This file contains evaluation utilities for referring expressions.
+    I.e. "The X closer to the Y" that indicate a specific object
+    Metrics are proportion of IoUs above a threshold
+    We can measure this for
+        - identifying the correct object
+        - IDing a different object of the target's class
+        - Identfying a different object
+    For each of these we can also measure average IoU given that it was above
+        a threshold.
+"""
+import warnings
+from collections import defaultdict
+from typing import Dict, Optional, Sequence
+
+import numpy as np
+import pytorch3d
+import torch
+from pytorch3d.ops import box3d_overlap
+from terminaltables import AsciiTable
+from torch import Tensor
+
+from home_robot.utils.bboxes_3d import get_box_verts_from_bounds
+
+# Return:
+# - IoU and IoU @ thresh
+# - Against target
+# - Against target class (is it getting distracted by other objects of the same class)
+# - Against all objects (is it localizing any object?)
+
+
+@torch.no_grad()
+def get_stats_ious_at_thresh(
+    ious: Tensor,
```

Comment:
We can probably move a lot of these to home_robot/datasets/eval?

Reply:
I'm thinking:
- Would it make sense to move it within the home_robot/datasets/scannet/eval?
- Do we plan to evaluate other datasets in home_robot
-  Which would be easier to port out of home_robot?